toxcore/tox.c

```diff
@@ -2236,6 +2309,20 @@ void tox_self_get_dht_id(const Tox *tox, uint8_t *dht_id)
     }
 }
 
+void tox_set_av_object(Tox *tox, void *object)
+{
+    lock(tox);
+    tox->toxav_object = object;
+    unlock(tox);
+}
+
+void tox_get_av_object(const Tox *tox, void **object)
```

Comment:
This function should return the object-ptr, not set a pointer to the object-ptr.

Reply:
return and lock won't work together. that's why i did it this way